### ISSUE TITLE
Error: Argument "onNext" is not a valid function.

### ISSUE THREAD

**Issue by ljcremer** [codediodeio/geofirex#23]
I am running the following code that then gives me the errors below. Seems like it is happening in rxjs for some reason. It seems to be happening on the await get(query) call. 

async function getOffices() {
  const center = geofire.point(40.1, -119.1);
  const radius = 100;
  const field = 'geofirex';

  const query = geofire.collection('clients/Mondelez/datasets/default/offices').within(center, radius, field);
  const offices = await get(query)
}

Error: Argument "onNext" is not a valid function.
    at Validator.(anonymous function).values [as isFunction] (C:\Users\ljcremer\Google Drive\Development\TroopTravel\v2019.1\cloud-functions\functions\node_modules\firebase-admin\node_modules\@google-cloud\firestore\build\src\validate.js:99:27)
    at Query.onSnapshot (C:\Users\ljcremer\Google Drive\Development\TroopTravel\v2019.1\cloud-functions\functions\node_modules\firebase-admin\node_modules\@google-cloud\firestore\build\src\reference.js:1524:25)
    at Observable._subscribe (C:\Users\ljcremer\Google Drive\Development\TroopTravel\v2019.1\cloud-functions\functions\node_modules\geofirex\dist\index.cjs.js:1597:33)
    at Observable._trySubscribe (C:\Users\ljcremer\Google Drive\Development\TroopTravel\v2019.1\cloud-functions\functions\node_modules\rxjs\internal\Observable.js:44:25)
    at Observable.subscribe (C:\Users\ljcremer\Google Drive\Development\TroopTravel\v2019.1\cloud-functions\functions\node_modules\rxjs\internal\Observable.js:30:22)
    at MapOperator.call (C:\Users\ljcremer\Google Drive\Development\TroopTravel\v2019.1\cloud-functions\functions\node_modules\rxjs\internal\operators\map.js:32:23)
    at Observable.subscribe (C:\Users\ljcremer\Google Drive\Development\TroopTravel\v2019.1\cloud-functions\functions\node_modules\rxjs\internal\Observable.js:25:22)
    at C:\Users\ljcremer\Google Drive\Development\TroopTravel\v2019.1\cloud-functions\functions\node_modules\rxjs\internal\util\subscribeTo.js:22:31
    at Object.subscribeToResult (C:\Users\ljcremer\Google Drive\Development\TroopTravel\v2019.1\cloud-functions\functions\node_modules\rxjs\internal\util\subscribeToResult.js:10:45)
    at CombineLatestSubscriber._complete (C:\Users\ljcremer\Google Drive\Development\TroopTravel\v2019.1\cloud-functions\functions\node_modules\rxjs\internal\observable\combineLatest.js:76:46)
    at CombineLatestSubscriber.Subscriber.complete (C:\Users\ljcremer\Google Drive\Development\TroopTravel\v2019.1\cloud-functions\functions\node_modules\rxjs\internal\Subscriber.js:79:18)
    at Observable._subscribe (C:\Users\ljcremer\Google Drive\Development\TroopTravel\v2019.1\cloud-functions\functions\node_modules\rxjs\internal\util\subscribeToArray.js:8:20)
    at Observable._trySubscribe (C:\Users\ljcremer\Google Drive\Development\TroopTravel\v2019.1\cloud-functions\functions\node_modules\rxjs\internal\Observable.js:44:25)
    at Observable.subscribe (C:\Users\ljcremer\Google Drive\Development\TroopTravel\v2019.1\cloud-functions\functions\node_modules\rxjs\internal\Observable.js:30:22)
    at CombineLatestOperator.call (C:\Users\ljcremer\Google Drive\Development\TroopTravel\v2019.1\cloud-functions\functions\node_modules\rxjs\internal\observable\combineLatest.js:46:23)
    at Observable.subscribe (C:\Users\ljcremer\Google Drive\Development\TroopTravel\v2019.1\cloud-functions\functions\node_modules\rxjs\internal\Observable.js:25:22)
    at MapOperator.call (C:\Users\ljcremer\Google Drive\Development\TroopTravel\v2019.1\cloud-functions\functions\node_modules\rxjs\internal\operators\map.js:32:23)
    at Observable.subscribe (C:\Users\ljcremer\Google Drive\Development\TroopTravel\v2019.1\cloud-functions\functions\node_modules\rxjs\internal\Observable.js:25:22)
    at TakeSubscriber.shareReplayOperation (C:\Users\ljcremer\Google Drive\Development\TroopTravel\v2019.1\cloud-functions\functions\node_modules\rxjs\internal\operators\shareReplay.js:21:35)
    at Observable.subscribe (C:\Users\ljcremer\Google Drive\Development\TroopTravel\v2019.1\cloud-functions\functions\node_modules\rxjs\internal\Observable.js:25:22)
    at TakeOperator.call (C:\Users\ljcremer\Google Drive\Development\TroopTravel\v2019.1\cloud-functions\functions\node_modules\rxjs\internal\operators\take.js:38:23)
    at Observable.subscribe (C:\Users\ljcremer\Google Drive\Development\TroopTravel\v2019.1\cloud-functions\functions\node_modules\rxjs\internal\Observable.js:25:22)

**Comment by antonioallen** [comment]
I'm getting this as well

**Comment by WisdomSky** [comment]
I'm also experiencing this issue.

**Comment by mpujolg** [comment]
Same issue

**Comment by maustinstar** [comment]
I had the same problem. Roll back your rxjs version.
`npm uninstall rxjs`
`npm install rxjs@6.0.0`

**Comment by ovidiu-balau** [comment]
Having the same issue.

**Comment by jos1337** [comment]
I'm getting just empty results, not even an error. Any ideas?

**Comment by joseluismurillorios** [comment]
I had troubles with my node version.

Upgraded to 8.11.2

Clenaed cache

Installed dependencies:
  "firebase": "5.2.0",
  "firebase-admin": "^7.0.0",
  "geofirex": "0.0.6",

Initialize
`admin.initializeApp(adminconfig);

const settings = { timestampsInSnapshots: true };

const db = admin.firestore();
db.settings(settings);`

And everything works fine again.

**Comment by philipfeldmann** [comment]
@joseluismurillorios 
Are you running it in cloud functions? I am stilling getting that error when using cloud functions, even after updating.

**Comment by joseluismurillorios** [comment]
@philipfeldmann 
Sorry, I'm not using geofirex in cloud functions, but I found problems with heroku so I had to specify the node version in the package.json
```
 "engines": {
    "node": "8.11.2"
  },
  "dependencies": {
    "firebase": "5.2.0",
    "firebase-admin": "^7.0.0",
    "geofirex": "0.0.6",
    ...
  },
```

**Comment by juanmiret** [comment]
I'm having the same issue!! I think it is definitively related to this: https://github.com/firebase/firebase-admin-node/issues/466

I need help :S this is a very important part of my app

**Comment by kedavidsa** [comment]
I'm trying to use this in a firebase  Cloud Function and the error I get is: Value for argument "onNext" is not a valid function. Someone have any solution or workaround to solve this?

**Comment by borntodesign** [comment]
I have this as well

**Comment by ryanlangton** [comment]
same problem here.. using in a firebase cloud function

**Comment by davepaiva** [comment]
I'm facing the same issues... is it due to the migration rxJS to the newer API?

**Comment by fabidona** [comment]
I have this as well

**Comment by ryanlangton** [comment]
This is a problem when using firebase-admin because it doesn't implement the onNext function. If you don't need admin access, for example if you allow open reads like I do, you can use firebase which does work. Here's code that's working for me.

` 
  const firebase = require('firebase');
  const geofirex = require('geofirex');

  var app;
  if (!firebase.apps.length) {
    app = firebase.initializeApp(config);
  }
  geofirex.init(app);
`

**Comment by kedavidsan** [comment]
@ryanlangton are you using that code inside an firebase cloud function? It is not working for me. I'm getting:
`TypeError: app.firestore is not a function
    at new GeoFireCollectionRef (/srv/node_modules/geofirex/dist/index.cjs.js:1465:24)
    at GeoFireClient.collection (/srv/node_modules/geofirex/dist/index.cjs.js:1639:16)
    at exports.positionUpdate.functions.firestore.document.onCreate (/srv/index.js:88:23)
    at cloudFunctionNewSignature (/srv/node_modules/firebase-functions/lib/cloud-functions.js:114:23)
    at /worker/worker.js:825:24
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:229:7)`

**Comment by kedavidsa** [comment]
@ryanlangton are you using that code inside an firebase cloud function? It is not working for me. I'm getting:
`TypeError:app.firestore is not a function at new GeoFireCollectionRef (/srv/node_modules/geofirex/dist/index.cjs.js:1465:24) at GeoFireClient.collection (/srv/node_modules/geofirex/dist/index.cjs.js:1639:16) at exports.positionUpdate.functions.firestore.document.onCreate (/srv/index.js:88:23) at cloudFunctionNewSignature (/srv/node_modules/firebase-functions/lib/cloud-functions.js:114:23) at /worker/worker.js:825:24 at <anonymous> at process._tickDomainCallback (internal/process/next_tick.js:229:7)`

**Comment by kedavidsa** [comment]
Now it is working by using the firebase library 5.2.0, with the last version it is not working in firebase cloud functions.

**Comment by dsg38** [comment]
> I had troubles with my node version.
> 
> Upgraded to 8.11.2
> 
> Clenaed cache
> 
> Installed dependencies:
> "firebase": "5.2.0",
> "firebase-admin": "^7.0.0",
> "geofirex": "0.0.6",
> 
> Initialize
> `admin.initializeApp(adminconfig);`
> `const settings = { timestampsInSnapshots: true };`
> `const db = admin.firestore();`
> `db.settings(settings);`
> 
> And everything works fine again.

Hey @joseluismurillorios - which version of rxjs are you using?

**Comment by joseluismurillorios** [comment]
@dsg38 you can find my project dependencies [here](https://github.com/joseluismurillorios/medeqs/blob/master/package.json)

**Comment by rjburdish** [comment]
Hey guys, not sure if any of you are still having this issue - This issue seems to only appear when you are using the admin SDK. I was able to resolve this by using the firebase client SDK "firebase". 

Unfortunately, this may also require you to add read/write permissions. 

Once you initialize the firebase app, you may still get the error: TypeError: app.firestore is not a function - you can resolve this by importing the following

```
import { firebase } from "@firebase/app";

const app = firebase.initializeApp({/*config*/})
const db = geofirex.init(app);

```

**Comment by Johnyoat** [comment]
This is the solution guys
After a while, this help @rjburdish you should import firestore` import '@firebase/firestore'`

```
import { firebase } from '@firebase/app';
import '@firebase/firestore'
import { get } from 'geofirex';

const app = firebase.initializeApp({/*config*/})
const db = geofirex.init(app);
```

Returning the response will be like

```
get(db.within(center,radius,field)).then(data=>{
      response.send({
        code : 200,
        message : 'Successful',
        data : data
      })
    }).catch(err=>{
      response.send({
        code : 500,
        message : err,
      })
    })
```

**Comment by DustinJSilk** [comment]
Does anyone know what the best route to take would be to make this plugin work for both web clients and functions, from one install?

**Comment by DustinJSilk** [comment]
So I did some digging around and found that if I change line 202 in `collection.ts` from:

```JavaScript
const unsubscribe = input.onSnapshot(observer);
```
to this:
```JavaScript
const unsubscribe = input.onSnapshot((val) => observer.next(val), err => observer.error(err));
```
It seems to work okay.

**Comment by azurechen** [comment]
@DustinJSilk you really save my day

**Comment by codediodeio** [comment]
Fixed in #78, use release `>=0.1.0` for node support.
